.infra/index.ts

```diff
@@ -83,10 +85,32 @@ const envVars: Record<string, Input<string>> = {
   redisHost,
 };
 
+const deadLetterTopicName = 'api.v1.dead-letter'
+const deadLetterTopic = new Stream(deadLetterTopicName, { isAdhocEnv, name: deadLetterTopicName, })
+
+const addDeadLetterTopicToWorkers = (workers: Worker[]): Worker[] => {
+  return workers.map(worker => {
+    if (worker.args?.deadLetterPolicy) {
+      return {
+        ...worker,
+        args: {
+            ...worker.args,
+            deadLetterPolicy: {
+              ...worker.args.deadLetterPolicy,
+              deadLetterTopic: deadLetterTopic.name.apply(name => name)
```

Comment:
It's the same. When you do apply without manipulation it has no effect

```suggestion
              deadLetterTopic: deadLetterTopic.name
```